src/coreclr/debug/daccess/request.cpp

```diff
@@ -2821,7 +2821,7 @@ ClrDataAccess::GetGCHeapStaticData(struct DacpGcHeapDetails *detailsData)
     {
         DPTR(dac_finalize_queue) fq = Dereference(g_gcDacGlobals->finalize_queue);
         DPTR(uint8_t*) fillPointersTable = dac_cast<TADDR>(fq) + offsetof(dac_finalize_queue, m_FillPointers);
-        for (unsigned int i = 0; i<(*g_gcDacGlobals->max_gen + 2 + dac_finalize_queue::ExtraSegCount); i++)
+        for (unsigned int i = 0; i < DAC_NUMBERGENERATIONS + 3; i++)
```

Comment:
Is it possible the copy now reads past the end of the source data? Previously these two loops differed by 2 (ExtraSegCount) and now they differ by 3. I didn't investigate what is correct from the GC standpoint, I am only observing it is different result than it was before.

Reply:
This is a weird beautiful double mistake, somehow, we are fine.

First of all, let's start with the truth, the correct answer is 5 + 2 = 7, `total_generation_count` is 0,1,2,LOH,POH = 5.
https://github.com/dotnet/runtime/blob/3bd0acf30587c88a1448a51a852871e3407aff5f/src/coreclr/gc/gcpriv.h#L4976-L4982

In the definition of we have the buffer size being 4 + 3 = 7
https://github.com/dotnet/runtime/blob/3bd0acf30587c88a1448a51a852871e3407aff5f/src/coreclr/inc/dacprivate.h#L711-L740

So in this particular case, it worked out, in a weird way.

The more interesting part is the `generation_table`, we will only fill 4 generations, and there is nothing we can do about that since `DacpGcHeapDetails` cannot be changed.

Reply:
cool, thanks for taking a look! 👍